Reflection/Login.simba

```diff
@@ -0,0 +1,299 @@
+    Exit(False);
+
+  //Adds a buffer to make sure that the world is completely clickable
+  WorldContainerBounds := [WorldContainer.Bounds.X1, WorldContainer.Bounds.Y1 + 15, WorldContainer.Bounds.X2, WorldContainer.Bounds.Y2 - 15];
+  WorldContainer.Free;
+
+  if WorldContainerBounds.Contains(R_WorldToBox(World)) then
+    Exit(True);
+
+  Direction := WorldContainerBounds.Middle.Y > R_WorldToBox(World).Middle.Y;
+
+  //Scroll the interface using the mouse-wheel
+  if ScrollType = TRSWorldScroll.SCROLLWHEEL then
+  begin
+    Mouse.Move(WorldContainerBounds);
+
+    T.Start;
+    while not WorldContainerBounds.Contains(R_WorldToBox(World)) do
+    begin
+      if T.ElapsedTime > 5000 then
+        Exit(False);
+
+      Mouse.Scroll(3, Direction);
+      writeln(R_WorldToBox(World));
+    end;
+
+    Exit(WorldContainerBounds.Contains(R_WorldToBox(World).Middle));
+  end;
+
+  //Scroll the interface using the scroll-bar
+  if ScrollType = TRSWorldScroll.SCROLLBAR then
+  begin
+    ScrollBar := RSWidget.Get(R_WORLD_SWITCHER_SCROLL_BAR.Group, R_WORLD_SWITCHER_SCROLL_BAR.Child, R_WORLD_SWITCHER_SCROLL_BAR.Index);
+    if ScrollBar.ref = nil then
+      Exit(False);
+
+    Mouse.Move(ScrollBar.Bounds);
+    Mouse.Click(MOUSE_LEFT);
+    Mouse.Hold(MOUSE_LEFT);
+    ScrollBar.Free;
+
+    T.Start;
+    while not WorldContainerBounds.Contains(R_WorldToBox(World)) do
+    begin
+      if T.ElapsedTime > 5000 then
+      begin
+        Mouse.Release(MOUSE_LEFT);
+        Exit(False);
+      end;
+
+      P := Mouse.GetPosition;
+      if Direction then
+        Mouse.Move(P.X + RandomRange(-1, 1), P.Y - RandomRange(3, 7))
+      else
+        Mouse.Move(P.X + RandomRange(-1, 1), P.Y + RandomRange(3, 7));
+    end;
+
+    Mouse.Release(MOUSE_LEFT);
+    Exit(WorldContainerBounds.Contains(R_WorldToBox(World).Middle));
+  end;
+
+  //Scroll the interface using the arrows
+  if ScrollType = TRSWorldScroll.SCROLLARROWS then
+  begin
+    if Direction then
+      ScrollArrow := RSWidget.Get(R_WORLD_SWITCHER_SCROLL_UP.Group, R_WORLD_SWITCHER_SCROLL_UP.Child, R_WORLD_SWITCHER_SCROLL_UP.Index)
+    else
+      ScrollArrow := RSWidget.Get(R_WORLD_SWITCHER_SCROLL_DOWN.Group, R_WORLD_SWITCHER_SCROLL_DOWN.Child, R_WORLD_SWITCHER_SCROLL_DOWN.Index);
+
+    if ScrollArrow.ref = nil then
+      Exit(False);
+
+    Mouse.Move(ScrollArrow.Bounds);
+    Mouse.Click(MOUSE_LEFT);
+    Mouse.Hold(MOUSE_LEFT);
+    ScrollArrow.Free;
+
+    T.Start;
+    while not WorldContainerBounds.Contains(R_WorldToBox(World)) do
+    begin
+      if T.ElapsedTime > 20000 then
+      begin
+        Mouse.Release(MOUSE_LEFT);
+        Exit(False);
+      end;
+    end;
+
+    Mouse.Release(MOUSE_LEFT);
+    Exit(WorldContainerBounds.Contains(R_WorldToBox(World).Middle));
+  end;
+end;
+
+{*
+R_SelectWorld
+~~~~~~~~~~~~~~~~~~~~~~
+Clicks on the selected world if it is visible in the list.
+
+Example:
+    R_SelectWorld(500);
+*}
+Function R_SelectWorld(World: Int32): Boolean;
```

Comment:
I think it would be cooler if this actually scrolled to the world for you

Reply:
Yea, I kept it as a separate function so that I can write a function that will implement all of the logic required for a complete action with only one function call. I could try implementing it into this one too.